### PR TITLE
Fix dex pod reload on config change

### PR DESCRIPTION
**Pull request by knrt10**
closes: #1039
Signed-off-by: knrt10 <kautilya@kinvolk.io>
Co-authored-by: Mateusz Gozdek <mateusz@kinvolk.io>

### CODE REVIEW

**Review by surajssd — CHANGES_REQUESTED** [review]
Please write an e2e test to verify this behaviour.

**Comment by invidian** [comment]
>Please write an e2e test to verify this behaviour.
This might be complex, as our e2e test suite do not have access to cluster configuration to modify it... What is your idea to test this @surajssd ?

**Comment by knrt10** [comment]
@invidian I had a talk with Suraj offline and I think this should be like this.

- Wait for dex deployment
- Get annotation checksum hash from pod template
- Update config map
- Get new annotation and compare with old.
- If changed, this means it has restarted the pod

What do you think?

**Comment by invidian** [comment]
>What do you think? I am little confused. Can we update it via client-go?

It can only be updated via Helm render, as it's Helm template which generates the checksum.

**Comment by surajssd** [comment]
In that case we can have a render test locally, which verifies that the annotations change on each render.

File: pkg/components/dex/component_test.go [hunk 1]
```diff
@@ -104,3 +135,35 @@ component "dex" {
...
 	}
 }
+
+func TestDeploymentAnnotationHashChange(t *testing.T) {
+	deploymentArr := []string{}
+
+	for _, tc := range tests {
+		b, d := util.GetComponentBody(tc.hcl, name)
+		if d != nil {
+			t.Errorf("%s - Error getting component body: %v", tc.desc, d)
```
Comment (invidian) [inline pkg/components/dex/component_test.go:145]: I think `Fatalf` should be used in all cases?

**Review by invidian — CHANGES_REQUESTED** [review]
Looks OK, though I think `Fatalf` should be used instead of `Errorf`, as most of the errors will cause further code to panic.

**Review by invidian — APPROVED** [review]
Looks OK

**Review by surajssd — APPROVED** [review]

**Pull request merged**